src/core/core.controller.js

```diff
@@ -341,7 +341,7 @@ helpers.extend(Chart.prototype, /** @lends Chart */ {
 			var id = scaleOptions.id;
 			var scaleType = valueOrDefault(scaleOptions.type, item.dtype);
 
-			if (positionIsHorizontal(scaleOptions.position) !== positionIsHorizontal(item.dposition)) {
+			if (positionIsHorizontal(scaleOptions.position, scaleOptions.axis) !== positionIsHorizontal(item.dposition)) {
```

Comment:
could fall back to first letter of id here:
```js
scaleOptions.axis || id[0]
```

Reply:
Done